snapcraft/remote/git.py

```diff
@@ -181,23 +184,66 @@ def push_url(
             "Pushing %r to remote %r with refspec %r.", ref, stripped_url, refspec
         )
 
-        # Create a list of tags to push
+        # temporarily call git directly due to libgit2 bug that unable to push
+        # large repos using https. See https://github.com/libgit2/libgit2/issues/6385
```

Comment:
Will you also reference the snapcraft ticket?
https://github.com/snapcore/snapcraft/issues/4478

Reply:
I can add that too